plugins/plugin-chart-echarts/src/Graph/controlPanel.tsx

```diff
@@ -0,0 +1,241 @@
+const columnSelectControl = {
+  ...sharedControls.entity,
+  type: 'SelectControl',
+  multi: false,
+  freeForm: true,
+  default: null,
+  includeTime: false,
+  optionRenderer: (c: ColumnMeta) => <ColumnOption column={c} showType />,
+  valueRenderer: (c: ColumnMeta) => <ColumnOption column={c} />,
+  valueKey: 'column_name',
+  allowAll: true,
+  filterOption: ({ data: opt }: { data: ColumnMeta }, text: string) =>
+    (opt.column_name && opt.column_name.toLowerCase().indexOf(text.toLowerCase()) >= 0) ||
+    (opt.verbose_name && opt.verbose_name.toLowerCase().indexOf(text.toLowerCase()) >= 0),
+  promptTextCreator: (label: string) => label,
+  commaChoosesOption: false,
+};
+
+const sourceControl = {
+  name: 'source',
+  config: {
+    ...columnSelectControl,
+    clearable: false,
+    label: t('Source'),
+    description: t('Name of the source nodes'),
+  },
+};
+
+const targetControl = {
+  name: 'target',
+  config: {
+    ...columnSelectControl,
+    clearable: false,
+    label: t('Target'),
+    description: t('Name of the target nodes'),
+  },
+};
+
+const categoryControl = {
+  name: 'category',
+  config: {
+    ...columnSelectControl,
+    clearable: true,
+    label: t('Color By'),
+    description: t('Optional category for nodes of graph'),
+    validators: [],
+  },
+};
+
+export default {
+  controlPanelSections: [
+    sections.legacyRegularTime,
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        [sourceControl],
+        [targetControl],
+        ['metric'],
+        [categoryControl],
+        ['adhoc_filters'],
+        ['row_limit'],
+      ],
+    },
+    {
+      label: t('Chart Options'),
+      expanded: true,
+      controlSetRows: [
+        ['color_scheme'],
+        [<h1 className="section-header">{t('Legend')}</h1>],
+        [showLegendControl],
+        [legendTypeControl, legendOrientationControl],
+        [legendMarginControl, noopControl],
+        [<h1 className="section-header">{t('Layout')}</h1>],
+        [
+          {
+            name: 'layout',
+            config: {
+              type: 'SelectControl',
+              renderTrigger: true,
+              label: t('Graph Layout'),
+              default: DEFAULT_FORM_DATA.layout,
+              choices: formatSelectOptions([
+                ['force', t('Force')],
+                ['circular', t('Circular')],
+              ]),
+              description: t('Layout type of graph'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'draggable',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Enable node draging'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.draggable,
+              description: t('Whether to enable node dragging in force layout mode.'),
+            },
+          },
+          {
+            name: 'roam',
+            config: {
+              type: 'CheckboxControl',
```

Comment:
Since ECharts supports it, maybe change this to a `SelectControl` to give users the choice to set just `scale` or `move`, too?

```
GRAPH ROAMING
  - Disabled
  - Scale and Move
  - Scale only
  - Move only
```

Although I can understand how a checkbox can keep things simple so this is totally optional.

Reply:
added